trees/trees.go

```diff
@@ -159,12 +159,16 @@ func GetTree(ctx context.Context, s storage.AdminStorage, treeID int64, opts Get
 	ctx, spanEnd := spanFor(ctx, "GetTree")
 	defer spanEnd()
 	tree, ok := FromContext(ctx)
-	if !ok || tree.TreeId != treeID {
+	if !ok {
 		var err error
 		tree, err = storage.GetTree(ctx, s, treeID)
 		if err != nil {
 			return nil, err
 		}
+	} else if tree.TreeId != treeID {
```

Comment:
Is it worth removing the `else` and checking the ID regardless of whether it came from storage or the context?

Reply:
Done. Amusingly this meant another test case needed to be updated.